nameko/timer.py

```diff
@@ -49,26 +52,29 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-
-        sleep_time = self.interval
+        sleep_time = 0 if self.eager else self.interval
 
         while True:
             # sleep for `sleep_time`, unless `should_stop` fires, in which
             # case we leave the while loop and stop entirely
-            with Timeout(sleep_time, exception=False):
```

Comment:
Uff.. Yes, i understand

I replaced the eventlet.Timeout() by my solution because self.should_stop.wait() wasn't called correctly under "with Timeout(..):".

I will think of an other solution for solve the problem.
thx